momepy/functional/_dimension.py

```diff
@@ -157,3 +158,72 @@ def perimeter_wall(
     ].values
 
     return results
+
+
+def weighted_character(
+    values: np.ndarray | Series, areas: np.ndarray | Series, graph: Graph
```

Comment:
```suggestion
    y: NDArray[np.float_] | Series, area: NDArray[np.float_] | Series, graph: Graph
```
proper typing, no plural in args.